interfaces/cython/cantera/ctml2yaml.py

```diff
@@ -0,0 +1,1137 @@
+                        phase_attribs["skip-undeclared-third-bodies"] = True
+                    else:
+                        raise ValueError(
+                            "Undefined value '{}' for third_bodies skip "
+                            "parameter".format(skip_node.get("third_bodies"))
+                        )
+                this_reactions = self.get_reaction_array(rA_node)
+                if has_filter:
+                    section_name = "{}-reactions".format(phase_name)
+                    reactions.append({section_name: this_reactions["reactions"]})
+                else:
+                    reactions.append(this_reactions)
+            # The reactions list may be empty, don't include it if it is
+            if reactions:
+                phase_attribs["kinetics"] = kinetics_model
+                internal_source = "reactions" in reactions[0]
+                # If there is one reactionArray node, no reaction filter
+                # has been specified, and the reactions are all from
+                # within this file, the output should be reactions: all,
+                # so we use update. Otherwise, there needs to be a list
+                # of mappings.
+                if len(reactions) == 1 and not has_filter and internal_source:
+                    phase_attribs.update(reactions[0])
+                else:
+                    phase_attribs["reactions"] = reactions
+
+        state_node = phase.find("state")
+        if state_node is not None:
+            phase_state = FlowMap()
+            for prop in state_node:
+                property_name = self._state_properties_mapping[prop.tag]
+                if prop.tag in [
+                    "moleFractions",
+                    "massFractions",
+                    "coverages",
+                    "soluteMolalities",
+                ]:
+                    value = split_species_value_string(prop.text)
+                else:
+                    value = get_float_or_units(prop)
+                phase_state[property_name] = value
+            if phase_state:
+                phase_attribs["state"] = phase_state
+
+        std_conc_node = phase.find("standardConc")
+        if std_conc_node is not None:
+            phase_attribs["standard-concentration-basis"] = std_conc_node.get("model")
+
+        self.phase_attribs = phase_attribs
+
+    def get_species_array(self, speciesArray_node):
+        """Process a list of species from a speciesArray node."""
+        species_list = FlowList(
+            speciesArray_node.text.replace("\n", " ").strip().split()
+        )
+        datasrc = speciesArray_node.get("datasrc", "")
+        if datasrc == "#species_data":
+            return {"species": species_list}
+        elif datasrc.startswith("#"):
+            return {datasrc[1:]: species_list}
+        else:
+            filename, location = datasrc.split("#", 1)
+            name = str(Path(filename).with_suffix(".yaml"))
```

Comment:
I don't think this is possible to do in general without knowing what the value of `CANTERA_DATA` is (or where the built-in data files are located somehow). For instance, someone referencing `gri30.xml` wouldn't have a copy of that in their local directory, and at least the minimal install of the Python interface won't have the data files included (but people could still want to do this conversion). I suppose this will just have to be caveat emptor.